tests_and_analysis/performance_benchmarking/artifact_handling/copy_benchmark_artifacts.py

```
@@ -0,0 +1,228 @@
+
+    Returns
+    -------
+    int
+        The value coerced into being between the lower and upper bound.
+    """
+    return max(lower_bound, min(value, upper_bound))
+
+
+def coerce_range(performance_benchmarking_response_json: Dict,
+                 range_parsed: Union[None, List[int]]) -> Tuple[int, int]:
+    """
+    Take details of the builds and find the first and last build numbers.
+    Coerce the range parsed by the user into between those first and last build
+    numbers.
+
+    Parameters
+    ----------
+    performance_benchmarking_response_json : Dict
+        A json response from Jenkins with build details including firstBuild
+        and lastBuild
+    range_parsed : Union[None, List[int]]
+        If no range was parsed this will be None, if a range has been parsed
+        this will be a list of two integers.
+
+    Returns
+    -------
+    Tuple[int, int]
+        The first value is the range start point and the second is the range
+        end point.
+    """
+    # The first and last build numbers
+    largest_possible_range = (
+        performance_benchmarking_response_json["firstBuild"]["number"] - 1,
+        performance_benchmarking_response_json["lastBuild"]["number"]
+    )
+    if range_parsed:
+        # Force both build numbers to be within the boundaries
+        lower_build_number = coerce(
+            range_parsed[0],
+            largest_possible_range[0], largest_possible_range[1]
+        )
+        upper_build_number = coerce(
+            range_parsed[1],
+            largest_possible_range[0], largest_possible_range[1]
+        )
+        return lower_build_number, upper_build_number
+    else:
+        # When there is no range parsed copy all the builds
+        return largest_possible_range
+
+
+def write_to_file(directory: str, filename: str,
+                  performance_benchmarks_json: str):
+    # Check access
+    test_file = os.path.join(directory, "test.txt")
+    try:
+        open(test_file, "w+")
```

Rather than creating a specific test file it might be better just to try and write the data that we want and let it fail if we don't have permissions - hopefully that should give a helpful error message anyway